### PR TITLE
Improve AudioStreamSample documentation.

### PR DESCRIPTION
**Pull request by follower**
Improves documentation to:

 * Make the description style more consistent with: https://github.com/godotengine/godot/blob/edc9097bc1604f0bd5ea7136c518699e9d64a731/modules/stb_vorbis/doc_classes/AudioStreamOGGVorbis.xml#L3-L8

 * Be more accurate with regard to the fact this class does not *play* the audio data it only stores the data. (I'm not sure that "audio stream driver" is a very useful phrase for either class but at least it's consistent.)

I also added a note that it's possible to generate audio data in a non-real-time fashion as this wasn't readily apparent as discussed in https://github.com/godotengine/godot/issues/18135#issuecomment-481976825.

The actual wording is only intended as a starting point.

### CODE REVIEW

File: doc/classes/AudioStreamSample.xml
Comment (LikeLakers2) [inline doc/classes/AudioStreamSample.xml]: This replacement sentence sounds like a lot of complicated words that new people won't understand nor really care about. Perhaps `Stores audio data`, or something similar? 

Also, I'm not sure the brief description should be the same as the description (or just the first sentence from the description) -- while you do cite a place where the two are the same, I don't know if many of the other documentation descriptions do this. If the two were always supposed to be the same, then there would be no point to the brief description.

**Comment by follower** [comment]
Related:

> AudioStream is the base class of all audio emitting objects. AudioStreamPlayer binds onto an AudioStream to emit PCM data into an AudioServer which manages audio drivers.
>
> All audio resources require two audio based classes: AudioStream and AudioStreamPlayback. As a data container, AudioStream contains the resource and exposes itself to GDScript. AudioStream references its own internal custom AudioStreamPlayback which translates AudioStream into PCM data.
>
> -- <http://docs.godotengine.org/en/latest/development/cpp/custom_audiostreams.html>

File: doc/classes/AudioStreamSample.xml
Comment (mhilbrunner) [reply doc/classes/AudioStreamSample.xml]: I kind of agree. Most of the new info are implementation deatils, maybe move them to the bottom of the full description, but I'd keep them out of the brief description. Maybe keep it as "Plays audio." or "Plays audio, can loop." for now? Not sure.

Comment (follower) [reply doc/classes/AudioStreamSample.xml]: I agree with both of your points @LikeLakers2.

I think "Stores audio data" is more accurate than the existing comment while also being short.

As I discovered when reading the Custom AudioStream docs (mentioned below in PR thread), I think one point of confusion is due to the fact that the class *does* both store & play audio but only the storage part is directly exposed to people using the class. The playback functionality is (as far as I am aware) is only exposed to (and via) the AudioStreamPlayer.

So, for non-engine developers, "Stores audio data" seems like a correct high level summary.

The implementation detail that the class also makes it possible to play the audio seems like an aside that can be made near the end of a longer description.

Thus, a structure more like:

> Stores audio data.
>
> Stores audio data loaded from WAVE (`.wav`) format audio files which can be played with an AudioStreamPlayer.
>
> [Code example.]
>
> Advanced developers can also generate the PCM format data directly. [something something]
>
> [A note for engine developers that the "playback" functionality is also in this class but accessed via AudioStreamPlayer]

Incidentally, I think the icon of this node in the IDE is also affected by this confusion (especially when compared to the OGGVorbis node).

Comment (LikeLakers2) [reply doc/classes/AudioStreamSample.xml]: I personally think that description could work, but it might benefit from a couple changes. Maybe something like this?
> AudioStreamSample stores audio data loaded from .WAV files, and can be used to play back the sound through an AudioStreamPlayer (for background music) or AudioStreamPlayer2D/AudioStreamPlayer3D (for positional audio).
> 
> This class can also be used to store PCM audio data that is generated dynamically through code.

As the class is generally made and configured through the IDE's inspector, I don't really think a code example is needed here -- also, if a developer really needs to use code to manipulate this class, I would imagine the code needed would be pretty obvious.

I'm personally unsure about having the note for engine developers be in the documentation -- perhaps that might go better in the actual .cpp/.h file defining AudioStreamSample, where developers would already be looking?

Also, if I can also make another suggestion for the brief description (this could use a tiny bit of rewording, but I believe it gets across its function better while still being easy to understand):
> Stores audio data, which is either loaded from .WAV files or generated in code.

Comment (akien-mga) [reply doc/classes/AudioStreamSample.xml]: The brief and long descriptions should not be the same. Typically, the brief version should be simpler.

File: doc/classes/AudioStreamSample.xml
Comment (akien-mga) [inline doc/classes/AudioStreamSample.xml]: Don't add empty lines in the XML, each line break creates a new paragraph.

**Comment by akien-mga** [comment]
Changes are still needed to simplify the brief description and remove the empty lines.

File: doc/classes/AudioStreamSample.xml
Comment (follower) [reply doc/classes/AudioStreamSample.xml]: Removed empty line in in https://github.com/follower/godot/commit/c0bd96e0e45567ae2bc36cc9963dd28d0613d7fa

**Comment by follower** [comment]
@akien-mga Have updated the descriptions.

Any further modification suggestions from anyone or is it good enough for now? :)

**Comment by akien-mga** [comment]
That looks great, thanks! Could you squash the commits together? See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase).

**Comment by LikeLakers2** [comment]
Seems fine to me now. :)

**Comment by follower** [comment]
@akien-mga I can squash the commits--if it makes a difference to you for me to do the squashing--but I was wondering if there was a reason for you to not use the GitHub PR squash functionality: https://stackoverflow.com/questions/35948131/github-squash-commits-from-web-interface-on-pull-request-after-review-comments#36377634?

The reason I ask is that I just used the GitHub web UI to do these doc edits so if it's not necessary for me for me to do anything locally it would be convenient. :)

But I can do so if that's your preference.

**Comment by akien-mga** [comment]
We don't use the squash and merge feature of GitHub for a number of reasons (most of which are no deal breakers, but put together they make this feature sub optimal, so we prefer that contributors make the PR reflect what it's final merged state will be).

**Comment by akien-mga** [comment]
GitHub allows maintainers to force push on PR branches though, so I can rebase it for you if need be.

**Comment by akien-mga** [comment]
I forced pushed a squashed version, can be merged once CI passes.

**Review by akien-mga — APPROVED** [review]

**Comment by follower** [comment]
Replaced by pseudo-"squashed" https://github.com/godotengine/godot/pull/29310.

(I realise replacing the PR isn't ideal but hopefully it still works for your purposes this time.)

**Comment by follower** [comment]
Oh, lol. :D

Somehow I missed your comment at https://github.com/godotengine/godot/pull/27928#issuecomment-496940675. And we happened to be working on it simultaneously. :)

Thanks for sorting it.

**Comment by follower** [comment]
Sorry for the overlap. I'll leave you to handle it. :)

**Comment by akien-mga** [comment]
#29310 was merged, so closing.

**Pull request closed without merging**
